src/core/Renderer.js

```diff
@@ -334,10 +369,18 @@ export class Renderer {
                 this.enable(this.gl.DEPTH_TEST);
                 this.setDepthMask(true);
             }
+
+            // Same for stencil
+            if(this.stencil || (!target || target.stencil)) {
+                this.enable(this.gl.STENCIL_TEST);
+                this.setStencilMask(0xff)
+            }
+
             this.gl.clear(
                 (this.color ? this.gl.COLOR_BUFFER_BIT : 0) |
                     (this.depth ? this.gl.DEPTH_BUFFER_BIT : 0) |
-                    (this.stencil ? this.gl.STENCIL_BUFFER_BIT : 0)
+                    (this.stencil ? this.gl.STENCIL_BUFFER_BIT : 0) |
+                    ((this.stencil && this.depth) ? this.gl.STENCIL_BUFFER_BIT : this.gl.DEPTH_BUFFER_BIT)
```

Comment:
in the last case, wouldn't that mean `this.gl.DEPTH_BUFFER_BIT` is included twice? (line 381 and line 383?)

Reply:
very true. fixed!